AWSAuthSDK/Sources/AWSMobileClient/Internal/_AWSMobileClient.m

```diff
@@ -64,16 +64,31 @@ + (instancetype)sharedInstance {
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
         _sharedMobileClient = [[_AWSMobileClient alloc] init];
-        _sharedMobileClient.isInitialized = NO;
-        _sharedMobileClient.signInProviderConfig = nil;
-        AWSFacebookSignInProviderClass = NSClassFromString(@"AWSFacebookSignInProvider");
-        AWSGoogleSignInProviderClass = NSClassFromString(@"AWSGoogleSignInProvider");
-        AWSCognitoUserPoolsSignInProviderClass = NSClassFromString(@"AWSCognitoUserPoolsSignInProvider");
     });
     
     return _sharedMobileClient;
 }
 
+#pragma initializers
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        self.isInitialized = NO;
```

Comment:
Better not to use `self.isInitialized` and `self.signInProviderConfig` inside the init method. On init the object is not fully initialized and calling the accessors can cause side effects.

Reply:
Thanks, that's a good call! I changed it and will publish a new revision soon